packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3122,18 +3107,64 @@ export class StripeHelper extends StripeHelperBase {
 
     const productPaymentCycleOld = this.stripePlanToPaymentCycle(planOld);
 
+    const sameBillingCycle =
+      productPaymentCycleOld === baseDetails.productPaymentCycleNew;
+
+    // get next invoice details
+    const nextInvoice = upcomingInvoiceWithInvoiceItem
+      ? upcomingInvoiceWithInvoiceItem
+      : await this.previewInvoiceBySubscriptionId({
+          subscriptionId: subscription.id,
+        });
+
+    const { total: nextInvoiceTotal, currency: nextInvoiceCurrency } =
+      nextInvoice;
+
+    // https://stripe.com/docs/api/invoices/object#invoice_object-lines
+    // as per the Stripe docs, one of the individual line items includes
+    // new plan pricing and tax information without additional fees
+    const upcomingInvoiceLineItem =
+      nextInvoice.lines && nextInvoice.lines.data
+        ? nextInvoice.lines.data.find((line) => !line.proration)
+        : undefined;
+
+    // if same billing cycle and line item exists, add amount and tax for new total
+    // else return the next invoice total
+    const newTotalInCents =
+      sameBillingCycle &&
+      upcomingInvoiceLineItem &&
+      upcomingInvoiceLineItem.tax_amounts
+        ? upcomingInvoiceLineItem.amount +
+          upcomingInvoiceLineItem.tax_amounts[0].amount
```

Comment:
Unfortunately we can't assume that there's only 1 entry in tax_amounts. I believe Julian might have some examples of countries or states with multiple tax amounts.

Reply:
After talking with Julian, I tested this with different places that have multiple tax amounts and the result was always the same. I'm wondering if Stripe totals the tax amount(s)? Do you have an example I could look at where it's not working?

Reply:
Discussed this with you in Slack, but an example of a location with multiple tax amounts is Quebec. [This is the PR ](https://github.com/mozilla/fxa/pull/14684) that added support on the frontend.

Reply:
Thanks so much! Updated to sum all the tax_amounts and tests for both same and different billing cycles. Moving onto discounts!